wallet/chain/p/builder/builder.go

```diff
@@ -645,29 +698,12 @@ func (b *builder) NewImportTx(
 		)
 	}
 
-	var (
-		inputs       []*avax.TransferableInput
-		outputs      = make([]*avax.TransferableOutput, 0, len(importedAmounts))
-		importedAVAX = importedAmounts[avaxAssetID]
-	)
-	if importedAVAX > txFee {
-		importedAmounts[avaxAssetID] -= txFee
-	} else {
-		if importedAVAX < txFee { // imported amount goes toward paying tx fee
-			toBurn := map[ids.ID]uint64{
-				avaxAssetID: txFee - importedAVAX,
-			}
-			toStake := map[ids.ID]uint64{}
-			var err error
-			inputs, outputs, _, err = b.spend(toBurn, toStake, ops)
-			if err != nil {
-				return nil, fmt.Errorf("couldn't generate tx inputs/outputs: %w", err)
-			}
+	outputs := make([]*avax.TransferableOutput, 0, len(importedAmounts))
+	for assetID, amount := range importedAmounts {
+		if assetID == avaxAssetID {
+			continue
```

Comment:
This `continue` has the same effect as the `delete(importedAmounts, avaxAssetID)` before the loop in the old code, but that was only if `importedAVAX <= txFee` (the `else`). Is that deliberate, or should this `if` be updated?

Reply:
Setting the `excessAVAX` to a non-zero value is the equivalent of not deleting the entry from `importedAmounts`.

Essentially, what this is doing is:
1. Populate outputs for all the imported non-AVAX assets (with the value that was imported)
2. Remove the static fees from the imported AVAX
3. If there was more AVAX imported than needed to pay the static fee, set the excess to that amount (and spend will add the output for it)
4. If there was not enough AVAX imported to pay the static fee, set the amount to burn to that amount (and spend will consume additional inputs to pay the fee)